include/boost/beast/http/impl/basic_parser.ipp

```diff
@@ -625,102 +581,53 @@ parse_chunk_header(char const*& p0,
     chunk-ext-val  = token / quoted-string
 */
 
-    auto p = p0;
+    auto p = in;
     auto const pend = p + n;
-    char const* eol;
 
-    if(! (f_ & flagFinalChunk))
+    if(n < skip_ + 2)
     {
-        if(n < skip_ + 2)
-        {
-            BOOST_BEAST_ASSIGN_EC(ec, error::need_more);
-            return;
-        }
-        if(f_ & flagExpectCRLF)
-        {
-            // Treat the last CRLF in a chunk as
-            // part of the next chunk, so p can
-            // be parsed in one call instead of two.
-            if(! parse_crlf(p))
-            {
-                BOOST_BEAST_ASSIGN_EC(ec, error::bad_chunk);
-                return;
-            }
-        }
-        eol = find_eol(p0 + skip_, pend, ec);
-        if(ec)
-            return;
-        if(! eol)
-        {
-            BOOST_BEAST_ASSIGN_EC(ec, error::need_more);
-            skip_ = n - 1;
-            return;
-        }
-        skip_ = static_cast<
-            std::size_t>(eol - 2 - p0);
-
-        std::uint64_t size;
-        if(! parse_hex(p, size))
+        BOOST_BEAST_ASSIGN_EC(ec, error::need_more);
+        return;
+    }
+    if(f_ & flagExpectCRLF)
+    {
+        // Treat the last CRLF in a chunk as
+        // part of the next chunk, so p can
+        // be parsed in one call instead of two.
+        if(! parse_crlf(p))
         {
             BOOST_BEAST_ASSIGN_EC(ec, error::bad_chunk);
             return;
         }
-        if(size != 0)
-        {
-            if (body_limit_.has_value())
-            {
-                if (size > *body_limit_)
-                {
-                    BOOST_BEAST_ASSIGN_EC(ec, error::body_limit);
-                    return;
-                }
-                *body_limit_ -= size;
-            }
-            auto const start = p;
-            parse_chunk_extensions(p, pend, ec);
-            if(ec)
-                return;
-            if(p != eol -2 )
-            {
-                BOOST_BEAST_ASSIGN_EC(ec, error::bad_chunk_extension);
-                return;
-            }
-            auto const ext = make_string(start, p);
-            this->on_chunk_header_impl(size, ext, ec);
-            if(ec)
-                return;
-            len_ = size;
-            skip_ = 2;
-            p0 = eol;
-            f_ |= flagExpectCRLF;
-            state_ = state::chunk_body;
-            return;
-        }
-
-        f_ |= flagFinalChunk;
     }
-    else
+
+    auto const eol = find_eol(p + skip_, pend, ec);
```

Comment:
@vinniefalco Now that we've removed `maybe_need_more`, I'm wondering if it's still worth keeping `skip_`. It's only used in this one place.

Reply:
hard for me to say. you will need to decide that

Reply:
It's pessimization when the chunk header can be read in one go, which is indeed the case most of the time because the chunk header is much smaller than the read buffer. I'd prefer to remove it and simplify the code flow and state.